tests/test_githubql.py

```diff
@@ -0,0 +1,528 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2015-2020 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Valerio Cosentino <valcos@bitergia.com>
+#
+
+import datetime
+import os
+import unittest.mock
+
+import httpretty
+import pkg_resources
+
+pkg_resources.declare_namespace('perceval.backends')
+
+from perceval.client import RateLimitHandler
+from perceval.utils import DEFAULT_DATETIME
+from perceval.backends.core.githubql import (logger,
+                                             GitHubQL,
+                                             GitHubQLCommand,
+                                             GitHubQLClient,
+                                             CATEGORY_EVENT,
+                                             MAX_CATEGORY_ITEMS_PER_PAGE)
+from base import TestCaseBackendArchive
+
+
+GITHUB_API_URL = "https://api.github.com"
+GITHUB_API_GRAPHQL_URL = GITHUB_API_URL + "/graphql"
+GITHUB_RATE_LIMIT = GITHUB_API_URL + "/rate_limit"
+GITHUB_REPO_URL = GITHUB_API_URL + "/repos/zhquan_example/repo"
+GITHUB_ISSUES_URL = GITHUB_REPO_URL + "/issues"
+GITHUB_ENTERPRISE_URL = "https://example.com"
+GITHUB_ENTERPRISE_API_URL = "https://example.com/api/v3"
+GITHUB_ENTERPRISE_API_GRAPHQL_URL = GITHUB_ENTERPRISE_URL + "/api/graphql"
+GITHUB_ENTREPRISE_RATE_LIMIT = GITHUB_ENTERPRISE_API_URL + "/rate_limit"
+GITHUB_ENTREPRISE_REPO_URL = GITHUB_ENTERPRISE_API_URL + "/repos/zhquan_example/repo"
+GITHUB_ENTERPRISE_ISSUES_URL = GITHUB_ENTREPRISE_REPO_URL + "/issues"
+
+
+def read_file(filename, mode='r'):
+    with open(os.path.join(os.path.dirname(os.path.abspath(__file__)), filename), mode) as f:
+        content = f.read()
+    return content
+
+
+class TestGitHubQLBackend(unittest.TestCase):
```

Comment:
I only miss a test checking the pagination.

Reply:
Added, please check again